content/4_Gold/LIS.mdx

```diff
@@ -144,20 +154,40 @@ int find_lis(vector<int> a) {
 <JavaSection>
 
 ```java
-public static int find_lis(int[] a) {
+public static int findLis(int[] a) {
 	ArrayList<Integer> dp = new ArrayList<Integer>();
 	for (int i : a) {
 		int pos = Collections.binarySearch(dp, i);
-		if(pos < 0) pos = Math.abs(pos + 1);
-		if (pos == dp.size()) dp.add(i);
-		else dp.set(pos, i);
+		pos = pos < 0 ? Math.abs(pos + 1) : pos;
+		if (pos == dp.size()) {  // we can have a new, longer increasing subsequence!
+			dp.add(i);
+		} else {  // oh ok, at least we can make the ending element smaller
+			dp.set(pos, i);
+		}
 	}
 	return dp.size();
 }
 ```
 
 </JavaSection>
 
+<PySection>
+
+```py
+# list from typing module
```

Comment:
Do you think it's better to explicitly import these modules?

Reply:
C++ and Java didn't, so I'm not sure.

Reply:
Ok then it's fine